app/src/lib/stores/helpers/fetch-tags-to-push-memoized.ts

```diff
@@ -0,0 +1,73 @@
+import memoizeOne from 'memoize-one'
+import { fetchTagsToPush } from '../../git'
+import { Repository } from '../../../models/repository'
+import { IGitAccount } from '../../../models/git-account'
+import { IRemote } from '../../../models/remote'
+
+type MemoizedFetchTagsArguments = Parameters<typeof fetchTagsToMemoize>
+
+/**
+ * Memoized version of the fetchTagsToPush git method.
+ *
+ * This method will return a cached result of the tags array when called multiple
+ * times with the same arguments, this is done to avoid doing too many network
+ * requests.
+ *
+ * @param repository  - The repository in which to check for unpushed tags
+ * @param account     - The account to use when authenticating with the remote
+ * @param remote      - The remote to check for unpushed tags
+ * @param branchName  - The branch that will be used on the push command
+ * @param localTags   - The current list of local tags. This is only used for memoization purposes
+ */
+export const fetchTagsToPushMemoized = memoizeOne(fetchTagsToMemoize, ((
+  newArgs: MemoizedFetchTagsArguments,
+  lastArgs: MemoizedFetchTagsArguments
+) => {
+  // When forceFetch is true, we consider the arguments different to
+  // force a call to the original method.
+  if (newArgs[6].forceFetch) {
+    return false
+  }
+
+  return serializeArguments(newArgs) === serializeArguments(lastArgs)
+}) as any)
+
+/**
+ * Temporary function to use on the memoization to make typescript happy.
+ *
+ * @param repository     - The repository in which to check for unpushed tags
+ * @param account        - The account to use when authenticating with the remote
+ * @param remote         - The remote to check for unpushed tags
+ * @param branchName     - The branch that will be used on the push command
+ * @param _localTags     - The current list of local tags. This is only used for memoization.
+ * @param _currentTipSha - The sha of the current HEAD commit. This is only used for memoization.
+ * @param _options       - Pass {forceFetch: true} to disable the memoization.
+ */
+function fetchTagsToMemoize(
+  repository: Repository,
+  account: IGitAccount | null,
+  remote: IRemote,
+  branchName: string,
+  _localTags: ReadonlyArray<string>, // only used for cache invalidation.
+  _currentTipSha: string, // only used for cache invalidation.
+  _options: { forceFetch: boolean } // only used for cache invalidation.
+) {
+  return fetchTagsToPush(repository, account, remote, branchName)
+}
```

Comment:
I'm going to check whether it makes more sense to use a standard cache via a `Map` object instead of `memoizeOne` here...

Reply:
I've finally used a `Map`, since it gives us more flexibility to invalidate the cache and the code is simpler: f7e3244c504d12ef8b04d3198c4b4e6cbc55c1b4